service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageController.kt

```diff
@@ -0,0 +1,153 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.messaging.message
+
+import fi.espoo.evaka.shared.Paged
+import fi.espoo.evaka.shared.auth.AuthenticatedUser
+import fi.espoo.evaka.shared.auth.UserRole
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.domain.Forbidden
+import fi.espoo.evaka.shared.domain.HelsinkiDateTime
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.util.UUID
+
+data class UnreadResponse(val unreadCount: Int)
+
+@RestController
+@RequestMapping("/messages")
+class MessageController {
+
+    @GetMapping("/my-accounts")
+    fun getAccountsByUser(db: Database.Connection, user: AuthenticatedUser): List<MessageAccount> {
+        return db.read { it.getMessageAccountsForUser(user) }
+    }
+
+    @GetMapping("/threads")
+    fun getThreads(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @RequestParam accountId: UUID,
+        @RequestParam pageSize: Int,
+        @RequestParam page: Int,
+    ): Paged<MessageThread> {
+        if (!db.read { it.getMessageAccountsForUser(user) }.map { it.id }.contains(accountId))
+            throw Forbidden("User is not authorized to access the account")
+        return db.read { it.getMessagesByAccount(accountId, pageSize, page) }
+    }
+
+    @GetMapping("/unread")
```

Comment:
Nit: Something like `/unread-count` would probably be slightly more idiomatic.

Reply:
Changed as discussed.